### PR TITLE
SDLInvaders: new recipe

### PR DESCRIPTION
**Pull request by TheZeldakatze**
I patched sdl-invaders to work on Haiku.
x86 and x86_gcc2 use gcc, whilst all other architectures use clang, because for some reason gcc doesn't like to build the game in those architectures, I don't know why, so I added an if statement to select between compilers

And I somehow managed to mess up the branches a bit and added ccatch to the branch somehow. I had some fun with git reset and now the correct files should be in the pull request

### CODE REVIEW

**Comment by TheZeldakatze** [comment]
> you should avoid hard coding the directory make use of find directory if you can. See the Haikuports Wiki.
> […](#)

I now changed the patch to use find_directory to get the folder paths. Did I patch that correctly?

File: games-arcade/sdlinvaders/sdlinvaders-0.7.6.recipe [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+LICENSE="GNU GPL v2"
+REVISION="1"
+SOURCE_URI="https://master.dl.sourceforge.net/project/sdlinvaders/sdlinvaders/0.7.6/SDLInvaders-0.7.6.tar.bz2?viasf=1"
+CHECKSUM_SHA256="7646ce1d6f93e554f972f47529bcca01cb7228ccc6cd41fd5fa28d7e5e91fd75"
+SOURCE_FILENAME="SDLInvaders-0.7.6.tar.bz2"
+SOURCE_DIR="SDLInvaders"
+PATCHES="sdlinvaders-0.7.6.patchset"
+ADDITIONAL_FILES="sdlinvaders.rdef.in"
+
+ARCHITECTURES="all"
```
Comment (Begasus) [inline games-arcade/sdlinvaders/sdlinvaders-0.7.6.recipe:14]: Didn't check the recipe yet, but from your statement, clang doesn't work for 32bit, you are forcing a build with gcc2 here, that could be an issue, maybe marking x86_gcc2 as broken and add SECONDARY_ARCHITECTURES="x86" could solve this?

Comment (TheZeldakatze) [reply games-arcade/sdlinvaders/sdlinvaders-0.7.6.recipe:14]: > Didn't check the recipe yet, but from your statement, clang doesn't work for 32bit, you are forcing a build with gcc2 here, that could be an issue, maybe marking x86_gcc2 as broken and add SECONDARY_ARCHITECTURES="x86" could solve this?

It's actually the other way around, for some reason the game doesn't build with gcc11, which would create linking errors. At the same time though using clang in an x86 or x86_gcc2 enviroment will fail, but using gcc works. I have a feeling that it's actually a problem with gcc11, as it does compile with gcc8 and gcc2 without an error.

**Comment by scottmc** [comment]
> > you should avoid hard coding the directory make use of find directory if you can. See the Haikuports Wiki.
> > […](#)
> 
> I now changed the patch to use find_directory to get the folder paths. Did I patch that correctly?

Do it run, and does it use the directories as you expect?

For Haiku specific changes in the patch, you might wrap them in #ifdefs so that your patch can be sent upstream and it won't break the game on other platforms.

**Comment by TheZeldakatze** [comment]
> > > you should avoid hard coding the directory make use of find directory if you can. See the Haikuports Wiki.
> > > […](#)
> > 
> > 
> > I now changed the patch to use find_directory to get the folder paths. Did I patch that correctly?
> 
> Do it run, and does it use the directories as you expect?
> 
> For Haiku specific changes in the patch, you might wrap them in #ifdefs so that your patch can be sent upstream and it won't break the game on other platforms.

It still runs fine. I now put the Haiku-specific stuff in #ifdefs. But I don't think that it will ever be upstreamed because the last release of the game is on sourceforge since 2004

**Comment by scottmc** [comment]
> It still runs fine. I now put the Haiku-specific stuff in #ifdefs. But I don't think that it will ever be upstreamed because the last release of the game is on sourceforge since 2004

Looks good to me.  You might take a look at this similar one next https://github.com/toivjon/sdl2-space-invaders next.

File: games-arcade/sdlinvaders/sdlinvaders-0.7.6.recipe (hunk 1 above)
Comment (extrowerk) [reply games-arcade/sdlinvaders/sdlinvaders-0.7.6.recipe:14]: >For some reason the game doesn't build with gcc11, which would create linking errors.
Could you please post the error messages?

Comment (TheZeldakatze) [reply games-arcade/sdlinvaders/sdlinvaders-0.7.6.recipe:14]: [sdlinvaders-0.7.6-build-gcc11.txt](https://github.com/haikuports/haikuports/files/7711765/sdlinvaders-0.7.6-build-gcc11.txt)
> > For some reason the game doesn't build with gcc11, which would create linking errors.
> 
> Could you please post the error messages?

(The text file has the whole output of make, it contains quite a few compile warnings)

I already tried to use the --std=c90 flag, It didn't change anything

I ran this make CC=gcc on a nightly build.

I installed a r1beta3 x86_64 image on a virtual machine which comes with gcc8 and then it builds without a problem. But as soon as I ran SoftwareUpdater (which installs gcc11) this error occured again. That's also the reason why I thought that the x86_gcc4 version was compiling fine, I still had the old gcc version, it now doesn't build with clang, which doesn't find stddef.h, nor gcc11.

Comment (TheZeldakatze) [reply games-arcade/sdlinvaders/sdlinvaders-0.7.6.recipe:14]: /boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//ccvduMt1.o:/boot/home/Desktop/SDLInvaders/common.h:126: multiple definition of `the_fighter'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:126: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//ccvduMt1.o:/boot/home/Desktop/SDLInvaders/common.h:127: multiple definition of `shots'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:127: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//ccvduMt1.o:/boot/home/Desktop/SDLInvaders/common.h:128: multiple definition of `shots2'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:128: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//ccvduMt1.o:/boot/home/Desktop/SDLInvaders/common.h:129: multiple definition of `blockades'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:129: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//ccvduMt1.o:/boot/home/Desktop/SDLInvaders/common.h:130: multiple definition of `galiens'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:130: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//ccvduMt1.o:/boot/home/Desktop/SDLInvaders/common.h:136: multiple definition of `highscore'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:136: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//ccvduMt1.o:/boot/home/Desktop/SDLInvaders/common.h:138: multiple definition of `darkfont'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:138: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//cccx6K5W.o:/boot/home/Desktop/SDLInvaders/common.h:126: multiple definition of `the_fighter'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:126: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//cccx6K5W.o:/boot/home/Desktop/SDLInvaders/common.h:127: multiple definition of `shots'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:127: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//cccx6K5W.o:/boot/home/Desktop/SDLInvaders/common.h:128: multiple definition of `shots2'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:128: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//cccx6K5W.o:/boot/home/Desktop/SDLInvaders/common.h:129: multiple definition of `blockades'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:129: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//cccx6K5W.o:/boot/home/Desktop/SDLInvaders/common.h:130: multiple definition of `galiens'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:130: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//cccx6K5W.o:/boot/home/Desktop/SDLInvaders/common.h:136: multiple definition of `highscore'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:136: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//cccx6K5W.o:/boot/home/Desktop/SDLInvaders/common.h:138: multiple definition of `darkfont'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:138: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//ccefpLAw.o:/boot/home/Desktop/SDLInvaders/common.h:126: multiple definition of `the_fighter'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:126: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//ccefpLAw.o:/boot/home/Desktop/SDLInvaders/common.h:127: multiple definition of `shots'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:127: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//ccefpLAw.o:/boot/home/Desktop/SDLInvaders/common.h:128: multiple definition of `shots2'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:128: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//ccefpLAw.o:/boot/home/Desktop/SDLInvaders/common.h:129: multiple definition of `blockades'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:129: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//ccefpLAw.o:/boot/home/Desktop/SDLInvaders/common.h:130: multiple definition of `galiens'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:130: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//ccefpLAw.o:/boot/home/Desktop/SDLInvaders/common.h:136: multiple definition of `highscore'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:136: first defined here
/boot/system/develop/tools/bin/../lib/gcc/x86_64-unknown-haiku/11.2.0/../../../../x86_64-unknown-haiku/bin/ld: /tmp//ccefpLAw.o:/boot/home/Desktop/SDLInvaders/common.h:138: multiple definition of `darkfont'; /tmp//ccWVzUOg.o:/boot/home/Desktop/SDLInvaders/common.h:138: first defined here

**Comment by Begasus** [comment]
Merging, keep track on the buildmaster for issues, thanks!

**Comment by Begasus** [comment]
@TheZeldakatze build failed on both 32bit and 64bit:
32bit: https://build.haiku-os.org/buildmaster/master/x86_gcc2/logviewer.html?buildruns/2218/builds/77282.log
64bit: https://build.haiku-os.org/buildmaster/master/x86_64/logviewer.html?buildruns/6502/builds/76558.log

**Comment by TheZeldakatze** [comment]
> @TheZeldakatze build failed on both 32bit and 64bit: 32bit: https://build.haiku-os.org/buildmaster/master/x86_gcc2/logviewer.html?buildruns/2218/builds/77282.log 64bit: https://build.haiku-os.org/buildmaster/master/x86_64/logviewer.html?buildruns/6502/builds/76558.log

The build error on 64 Bit seems to be because I used the clang5 packet on my computer whilst the build server uses clang12.

I now found the reason why gcc11 did not work. In gcc10 `-frepo` is now disabled by default, manually adding it fixes the build error (https://gcc.gnu.org/gcc-10/changes.html). I'll do a new pull request to fix it.

**Comment by Begasus** [comment]
Thanks, on the next PR I'll make sure it builds on 32bit before a merge (and possibly on 64bit) :)

**Pull request merged**